qa/tests/src/org/jboss/jbossts/qa/Utils/JDBCProfileStore.java

```diff
@@ -31,126 +31,108 @@
+        loadProfile();
 
-		return Integer.parseInt((String) _profile.get(profileName + "_NumberOfDrivers"));
-	}
+        return Integer.parseInt((String) _profile.get(profileName + "_NumberOfDrivers"));
+    }
 
-	public static String driver(String profileName, int driverNumber)
-			throws Exception
-	{
-		loadProfile();
+    public static String driver(String profileName, int driverNumber) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_Driver" + driverNumber);
-	}
+        return (String) _profile.get(profileName + "_Driver" + driverNumber);
+    }
 
-	public static String databaseURL(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String databaseURL(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_DatabaseURL");
-	}
+        return (String) _profile.get(profileName + "_DatabaseURL");
+    }
 
-	public static String databaseUser(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String databaseUser(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_DatabaseUser");
-	}
+        return (String) _profile.get(profileName + "_DatabaseUser");
+    }
 
-	public static String databasePassword(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String databasePassword(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_DatabasePassword");
-	}
+        return (String) _profile.get(profileName + "_DatabasePassword");
+    }
 
-	public static String databaseDynamicClass(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String databaseDynamicClass(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_DatabaseDynamicClass");
-	}
+        return (String) _profile.get(profileName + "_DatabaseDynamicClass");
+    }
 
-	// new methods for jndi
-	public static String binding(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    // new methods for jndi
+    public static String binding(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_Binding");
-	}
+        return (String) _profile.get(profileName + "_Binding");
+    }
 
-	public static String databaseName(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String databaseName(String profileName) throws Exception {
+        loadProfile();
 
-		return (String) _profile.get(profileName + "_DatabaseName");
-	}
+        return (String) _profile.getOrDefault(profileName + "_DatabaseName", "");
+    }
 
-	public static String host(String profileName)
-			throws Exception
-	{
-		loadProfile();
+    public static String serviceName(String profileName) throws Exception {
```

Comment:
> I don't think a test has been added for this?

Could you be more specific? I might be wrong but I think that this does not need a unit test. I copied this function from `databaseName`, which does not have any unit test.

> [...] it doesn't seem to use it?

The `serviceName` is only used when `OracleDataSource` is instantiated. In fact, this is something that only Oracle needs (so far).

Reply:
Although you said there is no unit test for `databaseName`, databaseName is currently tested by way of what you could consider an "integration test" because the default config has `DB1_THIN_JNDI_DatabaseName` and `DB2_THIN_JNDI_DatabaseName`.

In terms of testing when introducing the new code in a generic sense we should make sure it is tested. More specifically in your case if a unit test is not appropriate, it could be decided to have an integration test that reads some version of the config with something like `DB1_THIN_JNDI_ServiceName` and `DB2_THIN_JNDI_ServiceName` in it?

Reply:
> > [...] it doesn't seem to use it?
> 
> The `serviceName` is only used when `OracleDataSource` is instantiated. In fact, this is something that only Oracle needs (so far).

To this part of the thread, put another way, although you have handling for serviceName in the Java code and so there is now potential for different behaviour which I can understand I don't actually see that the change would take effect as in the `default` `JDBCProfiles` (for example) there is no profile config ending `_ServiceName` that I can see? Is the config file provided directly in the environment rather than using the default one in this repo?

Reply:
> Although you said there is no unit test for `databaseName`, databaseName is currently tested by way of what you could consider an "integration test" because the default config has `DB1_THIN_JNDI_DatabaseName` and `DB2_THIN_JNDI_DatabaseName`.
> 
> In terms of testing when introducing the new code in a generic sense we should make sure it is tested. More specifically in your case if a unit test is not appropriate, it could be decided to have an integration test that reads some version of the config with something like `DB1_THIN_JNDI_ServiceName` and `DB2_THIN_JNDI_ServiceName` in it?

I am now doubting my ability to copy&paste :sweat_smile: 
Sorry...I think I will go through [my other branch](https://github.com/jmfinelli/narayana/commit/cb2ed746b0bba3ae30ea19c09dc18735055ca369) before `git push` again